ckanext/saml2auth/helpers.py

```diff
@@ -121,3 +121,24 @@ def get_site_domain_for_cookie():
     parsed_url = urlparse(site_url)
     host = parsed_url.netloc.split(':')[0]
     return host if '.' in host else None
+
+
+def get_saml2auth_login_button_text():
+    """
+    Returns the configured text for the SAML2 login button.
+    Default is 'SSO' if not configured.
+
+    This version only returns the provider name instead of "login with %(provider)s"
+    to avoid duplication in the login template.
+    """
+    text = toolkit.config.get('ckanext.saml2auth.login_button_text', 'SSO')
+
+    # Check if the text follows the pattern "login with X"
+    import re
+    match = re.match(r'login\s+with\s+(\w+)', text, re.IGNORECASE)
+
+    if match:
+        # Extract and return only the provider name
+        return match.group(1)
+
+    return text
```

Comment:
```suggestion
def get_saml2auth_login_button_text():
    """
    Returns the configured text for the SAML2 login button.
    Defaults to 'SSO' if not configured.
    """
    text = toolkit.config.get('ckanext.saml2auth.login_button_text', 'SSO')

    return text
```
Simpler is better

Reply:
Also, add the `ckanext.saml2auth.login_button_text` to the README to let users know they can change this ugly text